lib/route-recognizer.js

```diff
@@ -88,6 +88,8 @@ EpsilonSegment.prototype = {
 function parse(route, names, specificity) {
   // normalize route as not starting with a "/". Recognition will
   // also normalize.
+  route = decodeURIComponent(route);
```

Comment:
this would decode special URI characters before parsing, so an encoded slash would get parsed as a path separator, `decodeURIComponent` should not happen before parsing, `decodeURI` is for normalizing a URI so that all the *optionally* encoded characters are decoded.